raft/src/main/java/org/apache/kafka/raft/RaftUtil.java

```diff
@@ -190,19 +192,24 @@ public static VoteResponseData singletonVoteResponse(
         int leaderEpoch,
         int leaderId,
         boolean voteGranted,
+        boolean preVote,
         Endpoints endpoints
     ) {
+        VoteResponseData.PartitionData partitionData = new VoteResponseData.PartitionData()
+            .setErrorCode(partitionLevelError.code())
+            .setLeaderId(leaderId)
+            .setLeaderEpoch(leaderEpoch)
+            .setVoteGranted(voteGranted);
+        if (apiVersion >= 2) {
+            partitionData.setPreVote(preVote);
+        }
```

Comment:
I am pretty convinced that we should remove the preVote field from the response. What do you think?